radix-transactions/src/manifest/static_resource_movements/effect.rs

```diff
@@ -0,0 +1,1541 @@
+use super::*;
+use radix_common::prelude::*;
+use radix_engine_interface::blueprints::access_controller::*;
+use radix_engine_interface::blueprints::account::*;
+use radix_engine_interface::blueprints::consensus_manager::*;
+use radix_engine_interface::blueprints::identity::*;
+use radix_engine_interface::blueprints::locker::*;
+use radix_engine_interface::blueprints::package::*;
+
+pub trait StaticInvocationResourcesOutput {
+    fn output(
```

Comment:
(Implemented in https://github.com/radixdlt/radixdlt-scrypto/pull/1931)

Reply:
> It's weird that they need to pass the instruction index - this could be handled in the layer above to turn it into uncertainty.

The client doesn't need to pass in an instruction index. Not entirely sure what this means?

> Having a Vec<InvocationIo> feels a little overlapping perhaps?

In what way?

> (Returning to this later) - in general, these APIs give too much access into their internals, which let `` shoot ourselves in the foot

I think examples of this would be useful. I think that we provide clients with a good amount of information that they can then use to do many things. So, I would like more information on this concern.

Looking at the code block above, I can see that the interface is nicer. However, I'm still not clear on why we need to make this change.

Reply:
> The client doesn't need to pass in an instruction index. Not entirely sure what this means?

The API for output mentions a `node_id` and `instruction_index`. I've changed this in 1931 to a parameter object refactor to avoid the method signature of `output` from being so long; and encapsulated `instruction_index` into a `change_source` object.